controller/share_manager_controller.go

```diff
@@ -923,6 +1119,13 @@ func (c *ShareManagerController) createShareManagerPod(sm *longhorn.ShareManager
 		}
 	}
 
+	staleNode := c.staleNodeMap[sm.Name]
+	if staleNode != "" {
```

Comment:
We refactored the logic so that controller never need to remember the staleNodeMap. It just looks at the current state of the lease CR to make decision